sql/core/src/main/scala/org/apache/spark/sql/execution/command/DescribeRelationJsonCommand.scala

```diff
@@ -0,0 +1,313 @@
+import org.apache.spark.sql.catalyst.expressions.{Attribute, AttributeReference}
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.catalyst.util.quoteIfNeeded
+import org.apache.spark.sql.connector.catalog.CatalogV2Implicits._
+import org.apache.spark.sql.connector.catalog.V1Table
+import org.apache.spark.sql.errors.QueryCompilationErrors
+import org.apache.spark.sql.types._
+import org.apache.spark.sql.util.PartitioningUtils
+
+/**
+ * The command for `DESCRIBE ... AS JSON`.
+ */
+case class DescribeRelationJsonCommand(
+    child: LogicalPlan,
+    partitionSpec: TablePartitionSpec,
+    isExtended: Boolean,
+    override val output: Seq[Attribute] = Seq(
+      AttributeReference(
+        "json_metadata",
+        StringType,
+        nullable = false,
+        new MetadataBuilder().putString("comment", "JSON metadata of the table").build())()
+    )) extends UnaryRunnableCommand {
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val jsonMap = mutable.LinkedHashMap[String, JValue]()
+    child match {
+      case v: ResolvedTempView =>
+        if (partitionSpec.nonEmpty) {
+          throw QueryCompilationErrors.descPartitionNotAllowedOnTempView(v.identifier.name())
+        }
+        describeIdentifier(Seq("system", "session", v.identifier.name()), jsonMap)
+        describeColsJson(v.metadata.schema, jsonMap)
+        describeFormattedTableInfoJson(v.metadata, jsonMap)
+
+      case v: ResolvedPersistentView =>
+        if (partitionSpec.nonEmpty) {
+          throw QueryCompilationErrors.descPartitionNotAllowedOnView(v.identifier.name())
+        }
+        describeIdentifier(v.identifier.toQualifiedNameParts(v.catalog), jsonMap)
+        describeColsJson(v.metadata.schema, jsonMap)
+        describeFormattedTableInfoJson(v.metadata, jsonMap)
+
+      case ResolvedTable(catalog, identifier, V1Table(metadata), _) =>
+        describeIdentifier(identifier.toQualifiedNameParts(catalog), jsonMap)
+        val schema = if (metadata.schema.isEmpty) {
+          // In older versions of Spark,
+          // the table schema can be empty and should be inferred at runtime.
+          sparkSession.table(metadata.identifier).schema
+        } else {
+          metadata.schema
+        }
+        describeColsJson(schema, jsonMap)
+        describeClusteringInfoJson(metadata, jsonMap)
+        if (partitionSpec.nonEmpty) {
+          // Outputs the partition-specific info for the DDL command:
+          // "DESCRIBE [EXTENDED|FORMATTED] table_name PARTITION (partitionVal*)"
+          describePartitionInfoJson(
+            sparkSession, sparkSession.sessionState.catalog, metadata, jsonMap)
+        } else {
+          describeFormattedTableInfoJson(metadata, jsonMap)
+        }
+
+      case _ => throw QueryCompilationErrors.describeAsJsonNotSupportedForV2TablesError()
+    }
+
+    Seq(Row(compact(render(JObject(jsonMap.toList)))))
+  }
+
+  private def addKeyValueToMap(
```

Comment:
The code below is moved from https://github.com/apache/spark/pull/49466/files#diff-632efff0ffceb2ff47b97f39570b4f2cff24401f132f559158acaa651b27af26L759